integration-tests/src/test/java/com/datastax/oss/driver/api/querybuilder/JsonInsertIT.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.querybuilder;
+
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.bindMarker;
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.insertInto;
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.selectFrom;
+import static com.datastax.oss.driver.assertions.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.core.cql.Row;
+import com.datastax.oss.driver.api.core.cql.SimpleStatement;
+import com.datastax.oss.driver.api.core.cql.Statement;
+import com.datastax.oss.driver.api.core.type.codec.CodecNotFoundException;
+import com.datastax.oss.driver.api.testinfra.ccm.CcmRule;
+import com.datastax.oss.driver.api.testinfra.session.SessionRule;
+import com.datastax.oss.driver.api.testinfra.session.SessionUtils;
+import com.datastax.oss.driver.categories.ParallelizableTests;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.time.Duration;
+import java.util.List;
+import java.util.Objects;
+import org.assertj.core.api.Assertions;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+
+@Category(ParallelizableTests.class)
+public class JsonInsertIT {
```

Comment:
Great coverage 👍 

I wonder if we need an integration test for this: the query builder only does text generation, so any part that requires a session here is not something that was impacted by this ticket. You could do all those checks in a unit test (building a `DefaultCodecRegistry` with your codec).

Integration tests take longer to run, so I think we should favor unit tests when we can.

Reply:
I've was examining that approach to test it similarly to `JsonInsertTest` but I think that unit test for `json()` will not give us as much assurance as of the `IT`.
A unit test is only testing the generated SQL. For json with codec variant, we still need to create some class (like `User`) and codec that is able to process it. 
If we are doing this the next step is to just write an integration test. When I was implementing that functionality it was very useful to have the `e-2-e` test for that. Also, we have no integration test for `JSON` so I think it will be beneficial to have one. In the `JsonInsertIT` I am using `ParallelizableTests` so it shares the same ccm with other tests - it is reasonably fast (on my mac ~1s for all tests. With already started ccm by other tests it will be faster). 
So I would like to leave this test as the `IT`, wdyt?

Reply:
OK, works for me.